### PR TITLE
Implement SDL-0237 Add Feature to Disable "Video Streaming Backgrounded String" Feature

### LINKED ISSUE #1304: [SDL 0237] Add Feature to Disable "Video Streaming Backgrounded String" Feature
Proposal: [Add Feature to Disable "Video Streaming Backgrounded String" Feature](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0237-add-feature-to-disable-video-streaming-backgrounded-string-feature.md)

> Add an option for app developer to disable the "Video Streaming Backgrounded String (SDL-0118)" feature.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/753

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with the following revision: rename `sendVideoBackgroundedString` to `showVideoBackgroundDisplay`.

> The proposal .md file was updated to reflect these revisions on 6/17/19.

### PR DESCRIPTION
**Pull request by t-yoshii**
Fixes #1304

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
We tested with our internal devboard

### Summary
Add an option for app developer to disable the "Video Streaming Backgrounded String (SDL-0118)" feature.

##### Breaking Changes
* Add parameter showVideoBackgroundDisplay to SDLStreamingMediaConfiguration class to disable the "video streaming backgrounded string (SDL-0118)" feature. The default value of this parameter is true and "Video Streaming Backgrounded String" will be sent (Current behaviour as of sdl_ios v6.2). If this parameter is set to false, backgroundingPixelBuffer will not be sent.

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLStreamingVideoLifecycleManager.m [hunk 1]
```diff
@@ -122,6 +123,7 @@ - (instancetype)initWithConnectionManager:(id<SDLConnectionManagerType>)connecti
     _useDisplayLink = configuration.streamingMediaConfig.enableForcedFramerateSync;
     _screenSize = SDLDefaultScreenSize;
     _backgroundingPixelBuffer = NULL;
+    _showVideoBackgroundDisplay = configuration.showVideoBackgroundDisplay;
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLStreamingVideoLifecycleManager.m]:
```suggestion
    _showVideoBackgroundDisplay = configuration.streamingMediaConfig.showVideoBackgroundDisplay;
```

Comment (t-yoshii) [reply SmartDeviceLink/SDLStreamingVideoLifecycleManager.m]: Thanks. Fixed.

Comment (t-yoshii) [reply SmartDeviceLink/SDLStreamingVideoLifecycleManager.m]: add test code in `SDLStreamingMediaConfigurationSpec`.

File: SmartDeviceLink/SDLStreamingMediaConfiguration.h [hunk 2]
```diff
@@ -85,6 +85,11 @@ typedef NS_ENUM(NSUInteger, SDLCarWindowRenderingType) {
  */
 @property (assign, nonatomic) BOOL allowMultipleViewControllerOrientations;
 
+/**
+ When YES, the StreamingMediaManager will send a black screen with "Video Backgrounded String". Defaults to YES.
+ */
+@property (assign, nonatomic) BOOL showVideoBackgroundDisplay;
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLStreamingMediaConfiguration.h]: @t-yoshii After more thought and using this, I think this approach won't work. It does not give apps the ability to modify if they show the background display after connection time, which is necessary to support requirements for all OEMs (some OEMs require the display, and others require that it not be there). I think we'll need to have a proposal revision to change this. Let me know if you'd like me to do it or if you'd like to do it yourself.

Comment (t-yoshii) [reply SmartDeviceLink/SDLStreamingMediaConfiguration.h]: Oh, you mean that the `showVideoBackgroundDisplay` is set at the initialization of lifecycle managers, however, HU type is known after RAI (later than initialization of lifecycle managers.)?
That's a good point.

I would like to hear your fix plan.

Comment (joeljfischer) [reply SmartDeviceLink/SDLStreamingMediaConfiguration.h]: I have submitted a proposal revision https://github.com/smartdevicelink/sdl_evolution/pull/796

Comment (t-yoshii) [reply SmartDeviceLink/SDLStreamingMediaConfiguration.h]: @joeljfischer 
I read your proposal and agree with the change. Thank you.

Comment (joeljfischer) [reply SmartDeviceLink/SDLStreamingMediaConfiguration.h]: Hello @t-yoshii, the proposal revision has been accepted! Would you like to make the changes in this PR, or submit a new PR, or I can submit a PR when I have time.

Comment (t-yoshii) [reply SmartDeviceLink/SDLStreamingMediaConfiguration.h]: @joeljfischer ,
I updated the PR to reflect the revision. Could you please review?

Thanks

**Pull request merged**